### PR TITLE
cool#9219 clipboard: support URLs in the payload of the setclipboard command

### PR DESCRIPTION
**Pull request by vmiklos**
Copying the full clipboard (including ODF) from a remote server into the
JS client failed with:

Refused to connect to '...' because it violates the following Content Security Policy directive: "connect-src ..."

If this happens, still send the original URL to the server, so it can
download the full clipboard, which gives better result for pasting,
compared to going via HTML only.

Not done in this commit: 1) the JS still tries to get the clipboard
itself, we download on the server only on failure and 2) the server does
a blocking HTTP download, which may not be wanted.

Signed-off-by: Miklos Vajna <vmiklos@collabora.com>
Change-Id: I9ed8c8050f9dbe71dc8fcd36c9d8fa6e8bba44f6

### CODE REVIEW

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04_cypress_multi_user/1254/ e.g. the multi-user calc/annotation_spec.js fails, but `make -C cypress_test check-multi spec=calc/annotation_spec.js` passes locally.

**Comment by vmiklos** [comment]
@caolanm could you please review this at some stage? Thanks.

If http fetch in the browser fails, send the URL to the server and fetch there, as that's a less locked down environment. Can be tested with a remote staging COOL vs local make run, copy on remote, paste on local. ODF paste is ~perfect if the copied content has e.g. a footnote and looks worse when we go via HTML.

File: wsd/ClientSession.cpp [hunk 1]
```diff
@@ -318,7 +318,59 @@ void ClientSession::handleClipboardRequest(DocumentBroker::ClipboardRequest
...
+                        const std::shared_ptr<const http::Response> httpResponse =
+                            session->response();
+                        if (httpResponse->statusLine().statusCode() != http::StatusCode::OK)
+                        {
+                            LOG_ERR("Clipboard download request failed");
+                            return;
+                        }
+
+                        std::string body = httpResponse->getBody();
+                        docBroker->forwardToChild(client_from_this(), "setclipboard\n" + body,
```
Comment (vmiklos) [inline wsd/ClientSession.cpp:346]: I'll make this more strict to require that the body is not a random format, but to some extent it's in the mimetype-length-data tuples format we expect here.

**Review by caolanm — APPROVED** [review]
lets merge work to date anyway

**Pull request merged**